plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcNamedRelationHandle.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.trino.spi.connector.SchemaTableName;
+
+import java.util.Objects;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class JdbcNamedRelationHandle
+        extends JdbcRelationHandle
+{
+    private final SchemaTableName schemaTableName;
+    private final RemoteTableName remoteTableName;
+
+    @JsonCreator
+    public JdbcNamedRelationHandle(
+            @JsonProperty("schemaTableName") SchemaTableName schemaTableName,
+            @JsonProperty("remoteTableName") RemoteTableName remoteTableName)
+    {
+        this.schemaTableName = requireNonNull(schemaTableName, "schemaTableName is null");
+        this.remoteTableName = requireNonNull(remoteTableName, "remoteTable is null");
+    }
+
+    @JsonProperty
+    public SchemaTableName getSchemaTableName()
+    {
+        return schemaTableName;
+    }
+
+    @JsonProperty
+    public RemoteTableName getRemoteTableName()
+    {
+        return remoteTableName;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        JdbcNamedRelationHandle that = (JdbcNamedRelationHandle) o;
+        return Objects.equals(schemaTableName, that.schemaTableName)
+                // remoteTableName is not compared here, as required by TestJdbcTableHandle#testEquivalence TODO document why this is important
```

Comment:
Can you explain it alrady in this PR?

Reply:
Sorry, missed this. I think i know, but @electrum would do this better (he authored the test)

Reply:
Fine. It was not documented previoulsly :)